src/cosl/coordinated_workers/nginx.py

```diff
@@ -29,6 +31,319 @@
     "nginx_port": 8080,
     "nginx_exporter_port": 9113,
 }
+RESOLV_CONF_PATH = "/etc/resolv.conf"
+
+
+class NginxLocationModifier(Enum):
+    """Enum representing valid Nginx `location` block modifiers."""
+
+    DEFAULT = ""  # prefix match
+    EXACT = "="  # exact match
+    REGEX = "~"  # case-sensitive regex
+
+
+@dataclass
+class NginxLocationConfig:
+    """Represents a `location` block in an Nginx configuration."""
+
+    upstream: str
+    path: str = "/"
```

Comment:
I'd make path mandatory (i.e. no default). it's good to force the user to be explicit about the most crucial arguments.